File: app/src/main/java/closer/vlllage/com/closer/api/models/GroupContactResult.java
package closer.vlllage.com.closer.api.models;

import closer.vlllage.com.closer.store.models.GroupContact;

public class GroupContactResult extends ModelResult {
    public String to;
    public String from;
    public PhoneResult phone;

    public static GroupContact from(GroupContactResult groupContactResult) {
        GroupContact groupContact = new GroupContact();
        groupContact.setId(groupContactResult.id);
        groupContact.setContactId(groupContactResult.from);
        groupContact.setGroupId(groupContactResult.to);
        groupContact.setContactName(groupContactResult.phone.name);
        groupContact.setContactActive(groupContactResult.phone.updated);
        groupContact.setUpdated(groupContactResult.updated);
        return groupContact;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/api/models/GroupContactResult.kt
package closer.vlllage.com.closer.api.models

import closer.vlllage.com.closer.store.models.GroupContact

class GroupContactResult : ModelResult() {
    var to: String? = null
    var from: String? = null
    var phone: PhoneResult? = null
    var group: GroupResult? = null
    var status: String? = null
    var inviter: String? = null
    var photo: String? = null

    companion object {

        fun from(groupContactResult: GroupContactResult): GroupContact {
            val groupContact = GroupContact()
            groupContact.id = groupContactResult.id
            groupContact.contactId = groupContactResult.from
            groupContact.groupId = groupContactResult.to
            groupContact.contactName = groupContactResult.phone!!.name
            groupContact.contactActive = groupContactResult.phone!!.updated
            groupContact.updated = groupContactResult.updated
            groupContact.status = groupContactResult.status
            groupContact.inviter = groupContactResult.inviter
            groupContact.photo = groupContactResult.photo
            return groupContact
        }
    }
}
